From 0de41e3adb5b506879cf149ef060c0e22843aae6 Mon Sep 17 00:00:00 2001
From: Evan Severson <evanseverson@google.com>
Date: Tue, 30 Nov 2021 18:19:18 -0800
Subject: [PATCH] Hide overlays on ReviewPermissionsAtivity

Test: atest PermissionReviewTapjackingTest
Bug: 176094367
Change-Id: I9f263b947853e14d081a73ce907917e9326b6ef7
(cherry picked from commit 7d5fd4443c5a2401dd0c0a50aaaaf6b2b3845374)
Merged-In: I9f263b947853e14d081a73ce907917e9326b6ef7
---
 .../permission/ui/ReviewPermissionsActivity.java              | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java b/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java
index 9d99bee6..a298a91f 100644
--- a/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java
+++ b/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java
@@ -21,6 +21,7 @@ import android.content.pm.PackageInfo;
 import android.content.pm.PackageManager;
 import android.os.Bundle;
 import android.text.TextUtils;
+import android.view.WindowManager;
 
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentActivity;
@@ -38,6 +39,9 @@ public final class ReviewPermissionsActivity extends FragmentActivity
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        getWindow().addSystemFlags(
+                WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         PackageInfo packageInfo = getTargetPackageInfo();
         if (packageInfo == null) {
             finish();
-- 
2.38.0.413.g74048e4d9e-goog

